compiler+runtime/src/jank/clojure/core.jank

```diff
@@ -3447,34 +3447,66 @@
        (def ~name ~expr))))
 
 ;; Case.
-(defn- shift-mask [shift mask x]
-  (-> x (bit-shift-right shift) (bit-and mask)))
-
 (def ^:private max-mask-bits 13)
 (def ^:private max-switch-table-size (bit-shift-left 1 max-mask-bits))
 (def ^:private int-min-value clojure.core-native/int-min)
 (def ^:private int-max-value clojure.core-native/int-max)
+(def ^:private int32-min-value clojure.core-native/int32-min)
+(def ^:private int32-max-value clojure.core-native/int32-max)
+
+(defn- shift-mask [shift mask x]
+  (-> x (bit-shift-right shift) (bit-and mask)))
+
+(defn- case-hash
+  "Returns the input if it is within the range of a 32-bit signed integer, otherwise returns the hash of the input."
+  [input]
+  (if (and (integer? input)
+           (>= input int32-min-value)
+           (<= input int32-max-value))
+    input
+    (hash input)))
 
 (defn- maybe-min-hash
   "Takes a collection of hashes and returns [shift mask] or nil if none found"
   [hashes]
   (first
-    (filter (fn [[s m]]
-              (apply distinct? (map #(shift-mask s m %) hashes)))
-            (for [mask (map #(dec (bit-shift-left 1 %)) (range 1 (inc max-mask-bits)))
-                  shift (range 0 31)]
-              [shift mask]))))
-
-(defn- case-map
+   (filter
+    (fn [[s m]]
+      (apply distinct? (map #(shift-mask s m %) hashes)))
+    (for [mask
+          (map #(dec (bit-shift-left 1 %)) (range 1 (inc max-mask-bits)))
+          shift
+          (range 0 31)]
+      [shift mask]))))
```

Comment:
Looks like the amount of code changes in this file would be cut by half without the whitespace changes. That would be easier to review.

Reply:
sorry about that. i formatted the code with cljfmt and seems we have no standard Clojure formatting yet.